File: ycsb/ethereum/src/main/java/site/ycsb/db/EthereumContractGasProvider.java
package site.ycsb.db;


import org.web3j.tx.gas.ContractGasProvider;

import java.math.BigInteger;

class EthereumContractGasProvider implements ContractGasProvider {
  @Override
  public BigInteger getGasPrice(String s) {
    return new BigInteger("0", 10);
  }

  @Override
  public BigInteger getGasPrice() {
    return new BigInteger("0", 10);
  }

  @Override
  public BigInteger getGasLimit(String s) {
    return new BigInteger("4700000", 10);
  }

  @Override
  public BigInteger getGasLimit() {
    return new BigInteger("4700000", 10);
  }
}

File: src/main/kotlin/connectionDetails/EthereumContractGasProvider.kt
package connectionDetails

import org.web3j.tx.gas.ContractGasProvider
import java.math.BigInteger

internal class EthereumContractGasProvider : ContractGasProvider {
    override fun getGasLimit(contractFunc: String?): BigInteger {
        return BigInteger("9000000", 10)
    }

    override fun getGasLimit(): BigInteger {
        return BigInteger("9000000", 10)
    }

    override fun getGasPrice(contractFunc: String?): BigInteger {
        return BigInteger("0", 10)
    }

    override fun getGasPrice(): BigInteger {
        return BigInteger("0", 10)
    }
}